text/0000-contract-metadata.md

```diff
@@ -0,0 +1,218 @@
+- Proposal Name: `contract_metadata`
+- Start Date: 2019-06-25
+- NEP PR: [nearprotocol/neps#0003](https://github.com/nearprotocol/NEPs/pull/3)
+
+# Summary
+[summary]: #summary
+
+This NEP introduces contract metadata, which summarizes the content of a given contract in json format.
+Contract metadata allows developers to easily list the methods in a contract and potentially display them to users to
+provide transparency. 
+It also provides the ability to generate bindings in different languages for the contract, thanks to the class information
+also contained in the metadata.
+
+# Motivation
+[motivation]: #motivation
```

Comment:
I think we need to also include app title in the metadata. 
If we look at playstore app submission process, it requires devs to fill in some display info for the app so that it can be listed in various UX contexts. For us, this might mean displaying the name of the app in the wallet for authorization, and/or NEAR app listing. This data may need to be localized if we want to support multiple languages in the future. 

For more info on play store app submission process see https://themanifest.com/app-development/how-publish-app-google-play-step-step-guide -- see product details section.

Reply:
This sounds very much to me as following on from <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md> With the idea of being able to add domain specific values (like app title) in a way that humans can verify it. Would it make sense to include all of this on contract, or out of band / on a separate part of the protocol?